src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs

```diff
@@ -313,11 +315,22 @@ private ICollection<FileInfo> PutFilesToRightModuleOrRoot(IEnumerable<ProjectDat
                     }
                 }
             }
-            return rootModuleFiles;
-
-            bool IsBinaryFile(FileInfo file) =>
-                file.Extension.Equals(".exe", StringComparison.InvariantCultureIgnoreCase)
-                || file.Extension.Equals(".dll", StringComparison.InvariantCultureIgnoreCase);
+            return new(rootModuleFiles, additionalFiles.Tests); // JS/TS does not use modules, so we don't need to put the test files in modules.
+
+            Dictionary<FileInfo, ProjectData[]> ProjectsPerFile(IEnumerable<ProjectData> projects) =>
+                projects
+                    .SelectMany(x => GetProjectFiles(x).Where(IsNotBinaryFile).Select(f => new { Project = x, File = f }))
+                    .GroupBy(x => x.File, FileInfoEqualityComparer.Instance)
+                    .ToDictionary(x => x.Key, x => x.Select(x => x.Project).ToArray());
+
+            IEnumerable<FileInfo> GetProjectFiles(ProjectData projectData) =>
+                projectData.ReferencedFiles
+                    .Concat(additionalFiles.Sources)
+                    .Except(additionalFiles.Tests, FileInfoEqualityComparer.Instance);
```

Comment:
We should add a comment on this line that the tests are removed since they are reported at the root level.